File: MatlabOctaveCodeSnippets/snip_signals_rect.m
%% Mimicking a continuous-time signal by using plot
Ts = 0.001; %sampling interval in seconds
t = -0.8:Ts:0.8; %discrete-time in seconds
pulse_width = 0.2; %support of this rect in seconds
tc = 0.4; %center (in seconds) of this rect
A = 2.5; %amplitude (in Volts) of this rect
x = A*rectpuls(t-tc,pulse_width);
subplot(211), plot(t,x) %plot as a continuos-time signal
xlabel('t'), ylabel('x(t)')
%% Making explicit the signal is discrete-time by using stem
n = -10:10; %discrete-time in seconds
pulse_width = 5; %support of this rect in samples
nc = -3; %center (in samples) of this rect
A = 7; %amplitude (in Volts) of this rect
x = A*rectpuls(n-nc,pulse_width);
subplot(212), stem(n,x) %plot as a discrete-time signal
xlabel('n'), ylabel('x[n]')

File: PythonCodeSnippets/snip_signals_rect.py
import numpy as np
import matplotlib.pyplot as plt

# Custom function to mimic MATLAB's rectpuls
def rectpuls(t, width):
    return np.where(np.abs(t) <= width / 2, 1, 0)

# Mimicking a continuous-time signal by using plot
Ts = 0.001  # Sampling interval in seconds
t = np.arange(-0.8, 0.8 + Ts, Ts)  # Discrete-time in seconds
pulse_width = 0.2  # Support of this rect in seconds
tc = 0.4  # Center (in seconds) of this rect
A = 2.5  # Amplitude (in Volts) of this rect
x = A * rectpuls(t - tc, pulse_width)

plt.subplot(2, 1, 1)
plt.plot(t, x)  # Plot as a continuous-time signal
plt.xlabel('t')
plt.ylabel('x(t)')

# Making explicit the signal is discrete-time by using stem
n = np.arange(-10, 11)  # Discrete-time in seconds
pulse_width = 5  # Support of this rect in samples
nc = -3  # Center (in samples) of this rect
A = 7  # Amplitude (in Volts) of this rect
x = A * rectpuls(n - nc, pulse_width)

plt.subplot(2, 1, 2)
plt.stem(n, x)  # Plot as a discrete-time signal
plt.xlabel('n')
plt.ylabel('x[n]')

plt.show()